src/PointwiseFunctions/GeneralRelativity/ComputeGhQuantities.cpp

```diff
@@ -196,19 +205,321 @@ Scalar<DataType> time_deriv_of_lapse(
+      spacetime_unit_normal, phi, pi);
+  return dt_shift;
+}
+
+namespace {
+template <size_t SpatialDim, typename Frame, typename DataType>
+struct D0LowerShiftBuffer;
+
+template <size_t SpatialDim, typename Frame>
+struct D0LowerShiftBuffer<SpatialDim, Frame, double> {
+  // NOLINTNEXTLINE(cppcoreguidelines-pro-type-member-init)
+  explicit D0LowerShiftBuffer(const size_t /*size*/) noexcept {}
+
+  tnsr::II<double, SpatialDim, Frame> inverse_spatial_metric;
+  tnsr::I<double, SpatialDim, Frame> dt_shift;
+  tnsr::ii<double, SpatialDim, Frame> dt_spatial_metric;
+};
+
+template <size_t SpatialDim, typename Frame>
+struct D0LowerShiftBuffer<SpatialDim, Frame, DataVector> {
+ private:
+  // We make one giant allocation so that we don't thrash the heap.
+  Variables<tmpl::list<::Tags::TempII<0, SpatialDim, Frame, DataVector>,
+                       ::Tags::TempI<1, SpatialDim, Frame, DataVector>,
+                       ::Tags::Tempii<2, SpatialDim, Frame, DataVector>>>
+      buffer_;
+ public:
+  explicit D0LowerShiftBuffer(const size_t size) noexcept
+      : buffer_(size),
+        inverse_spatial_metric(
+            get<::Tags::TempII<0, SpatialDim, Frame, DataVector>>(buffer_)),
+        dt_shift(get<::Tags::TempI<1, SpatialDim, Frame, DataVector>>(buffer_)),
+        dt_spatial_metric(
+            get<::Tags::Tempii<2, SpatialDim, Frame, DataVector>>(buffer_)) {}
+
+  tnsr::II<DataVector, SpatialDim, Frame>& inverse_spatial_metric;
+  tnsr::I<DataVector, SpatialDim, Frame>& dt_shift;
+  tnsr::ii<DataVector, SpatialDim, Frame>& dt_spatial_metric;
+};
+}  // namespace
+
+template <size_t SpatialDim, typename Frame, typename DataType>
+void time_deriv_of_lower_shift(
+    const gsl::not_null<tnsr::i<DataType, SpatialDim, Frame>*> dt_lower_shift,
+    const Scalar<DataType>& lapse,
+    const tnsr::I<DataType, SpatialDim, Frame>& shift,
+    const tnsr::ii<DataType, SpatialDim, Frame>& spatial_metric,
+    const tnsr::A<DataType, SpatialDim, Frame>& spacetime_unit_normal,
+    const tnsr::iaa<DataType, SpatialDim, Frame>& phi,
+    const tnsr::aa<DataType, SpatialDim, Frame>& pi) noexcept {
+  if (UNLIKELY(get_size(get<0>(*dt_lower_shift)) != get_size(get(lapse)))) {
+    *dt_lower_shift = tnsr::i<DataType, SpatialDim, Frame>(get(lapse));
+  }
+  // Use a Variables to reduce total number of allocations. This is especially
+  // important in a multithreaded environment.
+  D0LowerShiftBuffer<SpatialDim, Frame, DataType> buffer(get_size(get(lapse)));
+  // get \partial_0 N^j
+  buffer.inverse_spatial_metric =
+      determinant_and_inverse(spatial_metric).second;
+  buffer.dt_shift = time_deriv_of_shift<SpatialDim, Frame, DataType>(
+      lapse, shift, buffer.inverse_spatial_metric, spacetime_unit_normal, phi,
+      pi);
+  // Following loop to be replaced with:
+  // buffer.dt_spatial_metric =
+  //  GeneralizedHarmonic::time_deriv_of_spatial_metric<SpatialDim, Frame,
+  //    DataType>(lapse, shift, phi, pi);
```

Comment:
Do you intend to fix this before or after this PR is merged?  I guess the fix needs #1179?

Reply:
If #1179 gets merged before this is ready to be merged, then I will put in this fix before this PR is merged. However, ff this PR gets ready to be merged before #1179 , I strongly suggest we merge this PR without this fix. Afterward, when #1179 also gets merged I'll put in a quick patch for this fix.

Reply:
this is now fixed